app/controllers/api/v1/social_users_controller.rb

```diff
@@ -0,0 +1,55 @@
+module Api
+  module V1
+    class SocialUsersController < ApplicationController
+      before_action :authenticate_user!, only: %i[update]
+
+      # For social media logins, renders the appropriate `redirect_to` path depending on whether the user is registered.
+      #
+      # Requires that the ActionController::Parameters contain a :user key, with a nested :email key.
+      # For example: { user: { email: "john@example.com" } }
```

Comment:
Is this still what the params look like?  I ask b/c `params.dig(:email)` implies that `email` is no longer nested in a `user` hash.

Reply:
done, changed it to reflect the new input format.